src/lib/errors.rs

```diff
@@ -1,28 +1,73 @@
-use std::io::{stderr, Write};
+use std::fmt;
 
 use colored::*;
 use lazy_static::lazy_static;
 
-pub type UfofmtError = Box<dyn std::error::Error + Send + Sync + 'static>;
-pub type Result<T> = std::result::Result<T, UfofmtError>;
+pub(crate) type Result<T> = std::result::Result<T, UfofmtError>;
 
 lazy_static! {
     pub static ref ERROR_INDICATOR: ColoredString = "[ERROR]".red().bold();
     pub static ref OK_INDICATOR: ColoredString = "[OK]".green().bold();
 }
 
-pub(crate) fn print_error(err: &UfofmtError) {
-    let _ = writeln!(stderr(), "{} {}", *ERROR_INDICATOR, err);
+// ufofmt custom error type
+#[derive(Debug)]
+pub(crate) struct UfofmtError {
```

Comment:
If this is your main error type, convention would be to just call it `Error`. If someone wants to give it an alias when importing, they can do that. :)

Reply:
Ty!

Something along these lines?  2ac972e1e84381911534c20a873d3edcbb2972e8

Reply:
yea, exactly.

Also a matter of personal preference: once I have more than two fields I tend to stop using tuple variants and make them named fields, but this is personal. I definitely would go with named fields once you have more than two fields in a variant, though. :)